internal/bgp/frr/frr.go

```diff
@@ -261,23 +267,69 @@ func (sm *sessionManager) createConfig() (*frrConfig, error) {
 			} else if adv.Prefix.IP.To16() != nil {
 				version = "ipv6"
 			}
+			v4CommunityRuleIDs := map[string]int{}
+			v6CommunityRuleIDs := map[string]int{}
+
 			// Convert community 32bits value to : format
 			for _, c := range adv.Communities {
-				communities = append(communities, metallbconfig.CommunityToString(c))
+				community := metallbconfig.CommunityToString(c)
+				communities = append(communities, community)
+				ruleIdx++
+				v4CommunityRuleIDs[community] = ruleIdx
+				ruleIdx++
+				v6CommunityRuleIDs[community] = ruleIdx
+				addPrefixForCommunity(config, version, adv.Prefix.String(), community)
 			}
+			ruleIdx++
 			advConfig := advertisementConfig{
 				Version:     version,
 				Prefix:      adv.Prefix.String(),
 				Communities: communities,
 				LocalPref:   adv.LocalPref,
 			}
+			if adv.LocalPref != 0 {
+				addPrefixForLocalPref(config, version, adv.Prefix.String(), adv.LocalPref)
+			}
+
 			neighbor.Advertisements = append(neighbor.Advertisements, &advConfig)
 		}
 	}
 
 	return config, nil
 }
 
+func addPrefixForLocalPref(config *frrConfig, ipFamily, prefix string, localPref uint32) {
+	if ipFamily == "ipv4" {
```

Comment:
I was torn about this, because here Version is also what is gonna be rendered in the template, which only incidentally matches with the constant associated to ipfamily.IPv4 / v6, but I think I can accomodate the change.